packages/react/scroll-area/src/ScrollArea.tsx

```diff
@@ -944,9 +934,10 @@ function useDebounceCallback(callback: () => void, delay: number) {
 }
 
 function useResizeObserver(element: HTMLElement | null, onResize: () => void) {
-  // debounce to prevent error `ResizeObserver loop completed with undelivered notifications`
+  // debounce resize handler to ensure resizes in quick succession do not throw error:
+  // `ResizeObserver loop completed with undelivered notifications`
   const handleResize = useDebounceCallback(onResize, 10);
-  React.useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
Why does this need ton be `useLayoutEffect` and not just `useEffect`?

Reply:
because usually on resize you're calculating things from DOM dimensions and often updating UI to reflect those calculations. If it is a `useEffect` then you get the flicker.

I'm basically doing this:

> Use this to read layout from the DOM and synchronously re-render.
> https://reactjs.org/docs/hooks-reference.html#uselayouteffect

Reply:
ok, just a bit more careful about this since we discovered a few things the other day in terms of order etc. If it fixes something then makes sense.

Reply:
Yeah totally get it, I thought you might question this for that reason 🙂

Reply:
Just curious, what are the caveats you guys discovered?